database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -212,13 +244,67 @@ public void write(DataOutput out) throws IOException {
    * @param stmt the {@link PreparedStatement} to write the {@link StructuredRecord} to
    */
   public void write(PreparedStatement stmt) throws SQLException {
+    modifiableColumnTypes = new ArrayList<>(columnTypes);
+    operationName = operationName == null ? Operation.INSERT :
+      Operation.valueOf(operationName.toString().toUpperCase());
```

Comment:
Move this logic of using INSERT by default for operationName to a separate method getOperationName.